### ISSUE TITLE
Log message getting interpreted as Array and puking on strip!

### ISSUE THREAD

**Issue by bhudgens** [SumoLogic/fluentd-output-sumologic#65]
We have a log message that generates an odd error.  The message is shaped like this:

`[Container] 2020/11/18 20:07:50 Phase complete: BUILD State: SUCCEEDED`

The error we get:

![image](https://user-images.githubusercontent.com/7997948/99588437-2e239980-29b0-11eb-9ce6-798cb28d98bc.png)

We think it may be getting tripped up here:

https://github.com/SumoLogic/fluentd-output-sumologic/blob/7b25023c6e7443e4ddfb4cf48c7c8b675da32679/lib/fluent/plugin/out_sumologic.rb#L316-L319

I am unfamiliar with ruby enough to submit a bug fix or to know if the fix is appropriate.  Should that section be more explicit?

```ruby
          log = (record[@log_key]).to_s
          unless log.nil?
            log.strip!
          end
```

Lemme know if I can provide more context.

**Comment by cwebster2** [comment]
To add a bit of context on this, the stack trace points to the `log.strip!` on line 318 and this occurs whenever the message starts with a square bracket.

**Comment by sumo-drosiek** [comment]
@cwebster2 @bhudgens
Could you share configuration which leads to the given error? I'm curious how do you parse the given log record

**Comment by perk-sumo** [comment]
@cwebster2 @bhudgens which collection version you use?

**Comment by cwebster2** [comment]
Our configuration:

```
<source>
  @type http
  port  9090
  bind 0.0.0.0
  body_size_limit 32m
  keepalive_timeout 10s
  <parse>
    @type json
  </parse>
</source>

  <match aws.codebuild.*>
    @type rewrite_tag_filter
    <rule>
      key logGroup
      pattern /^\/aws\/codebuild\/[\d\w\-]+--.*$/
      tag gds.codebuild.logs
    </rule>
    <rule>
      key body
      pattern /.+/
      tag ignored.codebuild.logs
    </rule>
  </match>

  <match ignored.**>
    @type null
  </match>

  # extract the cluster and service name.
  <filter gds.**>
    @type parser
    key_name logGroup
    reserve_data true
    <parse>
      @type regexp
      expression /^\/aws\/codebuild\/(?<cluster>[\d\w\-]+)--(?<app>.*?)-+(?<action>(convert|build|terraform).*)$/
    </parse>
  </filter>

  <filter gds.**>
    @type parser
    key_name cluster
    reserve_data true
    <parse>
      @type regexp
      expression /^(v\d+-)?(?<clusterShort>.*?)(-v\d+)?$/i
    </parse>
  </filter>

  <match gds.**>
    @type sumologic
    endpoint "#{ENV["SUMOLOGIC_COLLECTOR_URL"]}"
    source_category gds/${clusterShort}/${app}/build
    source_name ${action}
    source_host ${logStream}
    add_timestamp false
    open_timeout 10
    log_format text
    log_key message
    <buffer clusterShort, app, action, logStream>
      #
      flush_interval 5s
      flush_thread_count 8
    </buffer>
  </match>
```

The messages being processed by this config look like:

```
{
    "timestamp": 1605670239700,
    "message": "[Container] Build phase complete.",
    "logStream": "ca6d652e-b4f2-4cc5-8002-cb28683c2918",
    "logGroup": "/aws/codebuild/v1-clustername--appname-convert-imagedetails-to-imagedefinitions"
}
```

This is using the `fluent/fluentd:v1.11-1` docker image as a base and installing `fluent-plugin-sumologic_output` and other gems on top of it.

**Comment by sumo-drosiek** [comment]
I'm able to reproduce the error, but not using given configuration and input data.

Here is my minimal config for reproduction of the error:

```
<source>
  @type dummy
  dummy {"timestamp": 1605670239700, "message": ["[Container] Build phase complete.", "some another element"], "logStream": "ca6d652e-b4f2-4cc5-8002-cb28683c2918", "logGroup": "/aws/codebuild/v1-clustername--appname-convert-imagedetails-to-imagedefinitions"}
  tag log.*
  rate 2000
</source>

<match log.**>
  @type sumologic
  endpoint "http://http:3001"
  add_timestamp false
  open_timeout 10
  log_format text
  log_key message
  <buffer>
    @type memory
  flush_interval "1s"
  </buffer>
</match>
```

And proposed PR to fix such cases: #66 

---

Anyway, for the given configuration I would check what comes from the input http plugin by using stdout filter:

```
<filter aws.codebuild.*>
  @type stdout
</filter>
```

If everything is correct (no array in the fluentd log) I would try to track what happen with records over full pipeline

**Comment by cwebster2** [comment]
Thanks for looking into this.  I spent some time yesterday trying to find the minimal case and like you i found i couldnt repro with the line i thought far triggering this.  I have stdout at every stage of the config and am trying to narrow down the actual offending line but your proposed PR looks like it will take care of our issue.

**Comment by sumo-drosiek** [comment]
@cwebster2 
Change has been released as [1.7.2](https://github.com/SumoLogic/fluentd-output-sumologic/releases/tag/1.7.2)

**Comment by cwebster2** [comment]
Thank you!
